integration_tests/utils.py

```diff
@@ -14,6 +14,44 @@ def create_test_cache() -> Generator[str, None, None]:
             yield d
 
 
+def list_files_recursively(path: str) -> List[str]:
+    """List all files recursively in the given path.
+    If the path is a file, return a list containing only that file.
+    If the path is a directory, list all files recursively in that directory."""
+    if os.path.isfile(path):
+        return [os.path.basename(path)]
+
+    files = []
+    for root, _, filenames in os.walk(path):
+        for filename in filenames:
+            # Create a relative path from the directory and add it to the list
+            rel_file = os.path.relpath(os.path.join(root, filename), path)
+            if not rel_file.startswith("./?"):
+                files.append(rel_file)
+    return sorted(files)
+
+
+def assert_files(path: str, expected_files: List[str]) -> bool:
+    """Assert that all expected files exist and are non-empty."""
+    files = list_files_recursively(path)
+    expected_files_sorted = sorted(expected_files)
+    if files != expected_files_sorted:
+        msg = "Files did not match expected"
```

Comment:
you could add a `test_case: unittest.TestCase` parameter and call `test_case.assertEqual(...)` to get nicer error message with diff and all.

Then, in your test you would call it by passing `self` as the first parameter.

```
assert_files(self, ...)
```